phonon-backend-vlc: use qt5 by default

Ma27 opened the pull request:
###### Motivation for this change

`minitube` is currently broken transitively due to the broken
`phonon-backend-qt4`: https://hydra.nixos.org/build/76523277

Although QT4 is fairly old, this package is still built with `qt4` ATM,
however QT5 is available as well. After this change, QT5 will be built
by default and in case anybody requires legacy QT4 it has to be enabled
explicitly like this:

```
with import <nixpkgs> { };
phonon-backend-vlc.override { withQt4 = true; }
```

Now the QT5-only build can be used (which fixes `minitube`) and there
are no confusions anymore with the build dependencies. Previously
`phonon-backend-vlc` and `libsForQt5.phonon-backend-vlc` used `qt4` by
default which was likely responsible for broken `minitube`.

The only package which uses `phonon-backend-vlc` as well is `tomahawk`
which is currently unmaintained in `nixpkgs` (see c76ad476b0245c474c057b451ba6fec18e7f81f1)
and broken since March (https://hydra.nixos.org/build/71873808).
FOr now I've marked it as broken as the last *stable* release is from
2015 (see https://github.com/tomahawk-player/tomahawk/releases/tag/0.8.4).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

xeji commented:
@GrahamcOfBorg build libsForQt5.phonon-backend-vlc minitube

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42586&attempt_id=d87fc046-8a88-4540-afc6-ea9e059895f3)

Attempted: libsForQt5.phonon-backend-vlc, minitube

<details><summary>Partial log (click to expand)</summary><p>

```
copying path '/nix/store/6chz9afwf66zybyczz7brn1iy9sx0g1y-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/6sc8isiky8jjf92xhimg1lxkazw6zhf4-qtsvg-5.11.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/jymlfasfc3hw77c8w5nwrsga67dy3nmq-qttools-5.11.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/zax3vyi4d0ay7sn5gxr0f46blf073ccr-qtx11extras-5.11.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/0wvm1dp8wxjgk87871mbb1f6jwv1r42a-qtxmlpatterns-5.11.0-dev' from 'https://cache.nixos.org'...
cannot build derivation '/nix/store/r2x9vpbk9xkvsjhx1n02ggnm04215qv4-vlc-3.0.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/mlzjr7rnsjdsgpcgiagwvn0hhgslgw18-phonon-backend-vlc-qt5-0.10.1.drv': 1 dependencies couldn't be built
copying path '/nix/store/rjmm769f84n236pzgp8pgdqhscyjnsns-qtdeclarative-5.11.0-dev' from 'https://cache.nixos.org'...
cannot build derivation '/nix/store/krph681cwzh7vcanywn5zk05ai8qf0gb-minitube-2.9.drv': 1 dependencies couldn't be built
error: build of '/nix/store/krph681cwzh7vcanywn5zk05ai8qf0gb-minitube-2.9.drv', '/nix/store/mlzjr7rnsjdsgpcgiagwvn0hhgslgw18-phonon-backend-vlc-qt5-0.10.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42586&attempt_id=7154e866-2d30-42a5-bc0b-50888c4d973a)

The following builds were skipped because they don't evaluate on x86_64-darwin: libsForQt5.phonon-backend-vlc, minitube

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42586&attempt_id=ba9005c5-8d8e-4c17-bfff-73b95b1d503a)

Attempted: libsForQt5.phonon-backend-vlc, minitube

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking RPATHs of ELF executables and libraries in /nix/store/zs9ggs31332gam82wzpzvbn3rbcbciz7-minitube-2.9
shrinking /nix/store/zs9ggs31332gam82wzpzvbn3rbcbciz7-minitube-2.9/bin/.minitube-wrapped
strip is /nix/store/4qvrxzxa535y8304mk195x50b6p9607d-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/zs9ggs31332gam82wzpzvbn3rbcbciz7-minitube-2.9/bin
patching script interpreter paths in /nix/store/zs9ggs31332gam82wzpzvbn3rbcbciz7-minitube-2.9
checking for references to /build in /nix/store/zs9ggs31332gam82wzpzvbn3rbcbciz7-minitube-2.9...
postPatchMkspecs
postPatchMkspecs
/nix/store/2kas4yj93ff48yy4z1ix6pabhsvj8lf0-phonon-backend-vlc-qt5-0.10.1
/nix/store/zs9ggs31332gam82wzpzvbn3rbcbciz7-minitube-2.9
```
</p></details>

xeji commented:
aarch64 failure: dependency `live555` has been broken for a while, unrelated to this PR.

xeji commented on pkgs/applications/audio/tomahawk/default.nix:
```diff
@@ -51,5 +51,6 @@ stdenv.mkDerivation rec {
     homepage = http://tomahawk-player.org/;
     license = licenses.gpl3Plus;
     platforms = platforms.all;
+    broken = true; # 2018-06-25
```
How is this related to phonon-backend-vlc ?
Also, please put this in a separate commit for a clean history.

Ma27 replied:
`tomahawk` was the only other package which depends on `phonon-backend-vlc`.
However it doesn't build anymore for some time (https://hydra.nixos.org/build/71873808) and I don't think it's worth to try making it compatible with my change as it's now unmaintained for quite a while according to c76ad476b0245c474c057b451ba6fec18e7f81f1)

Ma27 committed:
phonon-backend-vlc: use qt5 by default

`minitube` is currently broken transitively due to the broken
`phonon-backend-qt4`: https://hydra.nixos.org/build/76523277

Although QT4 is fairly old, this package is still built with `qt4` ATM,
however QT5 is available as well. After this change, QT5 will be built
by default and in case anybody requires legacy QT4 it has to be enabled
explicitly like this:

```
with import <nixpkgs> { };
phonon-backend-vlc.override { withQt4 = true; }
```

Now the QT5-only build can be used (which fixes `minitube`) and there
are no confusions anymore with the build dependencies. Previously
`phonon-backend-vlc` and `libsForQt5.phonon-backend-vlc` used `qt4` by
default which was likely responsible for broken `minitube`.

Ma27 committed:
tomahawk: mark as broken

During `8d6502f8ca9` `phonon-backend-vlc` uses QT5 by default.
Rather than fixing this it has been marked as broken as it's currently
unmaintained (see c76ad476b0245c474c057b451ba6fec18e7f81f1)
and broken since March (https://hydra.nixos.org/build/71873808).
For now I've marked it as broken as the last *stable* release is from
2015 (see https://github.com/tomahawk-player/tomahawk/releases/tag/0.8.4).

Ma27 commented:
@xeji created now two commits with a description and needed links to the affected Hydra builds. Is this fine for you now? %)

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42586&attempt_id=64d223ef-c8a6-4eb1-a627-97f1644e0b16)

Attempted: phonon-backend-vlc

The following builds were skipped because they don't evaluate on aarch64-linux: tomahawk

<details><summary>Partial log (click to expand)</summary><p>

```
builder for '/nix/store/fhs150gvirinppx9h31y9cb5md92b1vz-live555-2018.02.28.drv' failed with exit code 2
cannot build derivation '/nix/store/r2x9vpbk9xkvsjhx1n02ggnm04215qv4-vlc-3.0.3.drv': 1 dependencies couldn't be built
copying path '/nix/store/yw1rgbjzya6ib5j0rajz12g4vs1dhlxb-libmng-2.0.3' from 'https://cache.nixos.org'...
copying path '/nix/store/ymqllblhmk5lzqbi09n45va7by1k4np8-gst-plugins-base-0.10.36-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/y97hwkj6ir0kgbfynj37sjwbbk7vk93x-qt-4.8.7' from 'https://cache.nixos.org'...
copying path '/nix/store/jiqwxr6cgrjam0nbj7fwdqa6dyg8caxz-phonon-qt4-4.10.1' from 'https://cache.nixos.org'...
copying path '/nix/store/92b1k56fg7c04yp8n8vfgnz3j4irdiwk-qt-4.8.7-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/a43rjn46qvbcf1my04d562nslnpfybm0-phonon-qt4-4.10.1-dev' from 'https://cache.nixos.org'...
cannot build derivation '/nix/store/bq7nbhky0izpd8sdk52q14skkm05flaw-phonon-backend-vlc-qt4-0.10.1.drv': 1 dependencies couldn't be built
error: build of '/nix/store/bq7nbhky0izpd8sdk52q14skkm05flaw-phonon-backend-vlc-qt4-0.10.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42586&attempt_id=50b07816-c435-4aea-96fa-00f5973f5ec5)

Attempted: phonon-backend-vlc

The following builds were skipped because they don't evaluate on x86_64-linux: tomahawk

<details><summary>Partial log (click to expand)</summary><p>

```
  /nix/store/by6ig6vbh5mxi0c1vn0di8yzdbjw8cp1-phonon-qt4-4.10.1
  /nix/store/hyzrxs72s2jm8yrv4s5b1vqh9xhz7xbg-qt-4.8.7
  /nix/store/n24l6mq9sp8ihr74v50gcjwimbyc94dk-phonon-backend-vlc-qt4-0.10.1
  /nix/store/q4rp5lgzhxwvh1bd43dvjpmlm0vmhnwa-gst-plugins-base-0.10.36
copying path '/nix/store/05qw4g70cm9crmcjy40qx4kac5add2kb-gstreamer-0.10.36' from 'https://cache.nixos.org'...
copying path '/nix/store/q4rp5lgzhxwvh1bd43dvjpmlm0vmhnwa-gst-plugins-base-0.10.36' from 'https://cache.nixos.org'...
copying path '/nix/store/hyzrxs72s2jm8yrv4s5b1vqh9xhz7xbg-qt-4.8.7' from 'https://cache.nixos.org'...
copying path '/nix/store/by6ig6vbh5mxi0c1vn0di8yzdbjw8cp1-phonon-qt4-4.10.1' from 'https://cache.nixos.org'...
copying path '/nix/store/n24l6mq9sp8ihr74v50gcjwimbyc94dk-phonon-backend-vlc-qt4-0.10.1' from 'https://cache.nixos.org'...
/nix/store/n24l6mq9sp8ihr74v50gcjwimbyc94dk-phonon-backend-vlc-qt4-0.10.1
```
</p></details>

xeji commented:
Thanks :+1:

xeji commented:
@GrahamcOfBorg build libsForQt5.phonon-backend-vlc minitube

GrahamcOfBorg commented:
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42586&attempt_id=35fa43a3-2c73-4a63-ace1-c1483978de46)

The following builds were skipped because they don't evaluate on x86_64-darwin: libsForQt5.phonon-backend-vlc, minitube

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42586&attempt_id=748ae0c2-afa6-4884-a8c9-096c656757bd)

Attempted: libsForQt5.phonon-backend-vlc, minitube

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/2kas4yj93ff48yy4z1ix6pabhsvj8lf0-phonon-backend-vlc-qt5-0.10.1
/nix/store/zs9ggs31332gam82wzpzvbn3rbcbciz7-minitube-2.9
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42586&attempt_id=6b55f409-ddee-44b1-aed9-6aac09cf9912)

Attempted: libsForQt5.phonon-backend-vlc, minitube

<details><summary>Partial log (click to expand)</summary><p>

```
c++ -c -Iinclude -I../UsageEnvironment/include -I../groupsock/include -m64  -fPIC -I. -O2 -DSOCKLEN_T=socklen_t -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64 -Wall -DBSD=1 Media.cpp
g++: error: unrecognized command line option '-m64'
make[1]: *** [Makefile:32: Media.o] Error 1
make[1]: Leaving directory '/build/live/liveMedia'
make: *** [Makefile:33: all] Error 2
builder for '/nix/store/fhs150gvirinppx9h31y9cb5md92b1vz-live555-2018.02.28.drv' failed with exit code 2
cannot build derivation '/nix/store/r2x9vpbk9xkvsjhx1n02ggnm04215qv4-vlc-3.0.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/mlzjr7rnsjdsgpcgiagwvn0hhgslgw18-phonon-backend-vlc-qt5-0.10.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/krph681cwzh7vcanywn5zk05ai8qf0gb-minitube-2.9.drv': 1 dependencies couldn't be built
error: build of '/nix/store/krph681cwzh7vcanywn5zk05ai8qf0gb-minitube-2.9.drv', '/nix/store/mlzjr7rnsjdsgpcgiagwvn0hhgslgw18-phonon-backend-vlc-qt5-0.10.1.drv' failed
```
</p></details>
